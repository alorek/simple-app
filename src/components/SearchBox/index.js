import React, { Component } from 'react';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import debounce from 'lodash/debounce';

import { searchTable } from '../../actions';

class SearchBox extends Component {

    constructor(props) {
        super(props)
        this.handleSearch = debounce(this.handleSearch, 300).bind(this);
        this.onKeyUp = this.onKeyUp.bind(this);
    }

    onKeyUp(e) {
        this.handleSearch(e.target.value);
    }

    handleSearch(searchTerm) {
        this.props.searchTable(searchTerm);
    }

    render() {

        const inputStyle = {
            background: 'transparent',
            borderRight: 'none',
            boxShadow: 'inset 0 1px 0px rgba(0, 0, 0, .075)'
        };
        const addOnStyle = {
            background: 'transparent',

        };

        return (
            <div className="input-group">
                <input
                   onKeyUp={  this.onKeyUp  }
                   style={ inputStyle }
                    className="form-control"
                    placeholder="Search"
                    type="text"/>
                <span
                    className="input-group-addon"
                    style={ addOnStyle }>
                    <i className="glyphicon glyphicon-search"></i>
                </span>
            </div>
        );
    }

}

function mapDispatchToProps (dispatch) {
    return bindActionCreators({searchTable: searchTable}, dispatch);
}

function mapStateToProps(state) {
    return {
        searchTerm: state.table.searchTerm
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchBox);
