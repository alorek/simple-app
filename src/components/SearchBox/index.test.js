import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import reducer from '../../reducers';
import SearchBox from './index';

test('SearchBox renders correctly', () => {
    const store = createStore(reducer);

    const dataKeys = ['resource_type'];

    const tree = renderer.create(
        <Provider store={ store }>
            <Router>
                <SearchBox  />
            </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
