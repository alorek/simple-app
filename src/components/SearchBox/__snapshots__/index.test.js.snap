// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<div
  className="input-group"
>
  <input
    className="form-control"
    onKeyUp={[Function]}
    placeholder="Search"
    style={
      Object {
        "background": "transparent",
        "borderRight": "none",
        "boxShadow": "inset 0 1px 0px rgba(0, 0, 0, .075)",
      }
    }
    type="text"
  />
  <span
    className="input-group-addon"
    style={
      Object {
        "background": "transparent",
      }
    }
  >
    <i
      className="glyphicon glyphicon-search"
    />
  </span>
</div>
`;
